src/box/task/ThingsProcessingDemonstrator.java

```diff
@@ -0,0 +1,43 @@
+package box.task;
+
+import functionalinterface.example.Randomable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ThingsProcessingDemonstrator {
+
+    public static void demonstrateBoxTask() {
+        List<Box> boxes = Arrays.asList(
+                new Box(generateThings(100, "box one")),
+                new Box(generateThings(50, "box two")),
+                new Box(generateThings(75, "box three")));
+        List<Protector> protectedThings = new ArrayList<>();
+
+        boxes.forEach(box -> {
+            protectedThings.addAll(box.getThings().stream()
+                    .filter(Thing::isFragile)
+                    .map(Protector::new)
+                    .collect(Collectors.toList()));
+            box.getThings().removeIf(Thing::isFragile);
+        });
+
+
+        System.out.println("Not Fragile Things:" + boxes
+                + "\nProtected fragile things" + protectedThings);
+    }
+
+    private static List<Thing> generateThings(int howMany, String boxName) {
+        List<Thing> things = new ArrayList<>();
+        Randomable rand = (from, to) -> from + (int) (Math.random() * (to - from));
```

Comment:
```
Random random = new Random();
random.nextInt(11) + 1
```
or 

`(int) Math.random() * 11 + 1`

is much simpler and have better readability

Reply:
from + (int) (Math.random() * (to - from)); replaced by  (int) (Math.random() * (to - from)) + from ;